src/test/java/com/zaxxer/nuprocess/CatTest.java

```diff
@@ -382,9 +385,9 @@ int getExitCode()
 
       boolean checkAdlers()
       {
-         return readAdler32.getValue() == writeAdler32.getValue();
+         return writes == WRITES && readAdler32.getValue() == writeAdler32.getValue();
```

Comment:
Without this, if `onStdinReady` is _never called_, the `readAdler32` and `writeAdler32` values will match, since they're both empty. By checking the `writes` count we can ensure that not only do the values match, but they match after processing the expected number of writes.